### DISCUSSION TITLE
Multi-Animal Tracking Merge Projects

### DISCUSSION THREAD

**Discussion by djg9324** [talmolab/sleap#1557] [Help!] [answered]
I have been trying to sleap track 2 mice at the same time. I currently have two projects: 100 frames from the original video that I labeled the mice on (1) and 20 frames that I corrected after I ran predictions (2). After running predictions and editing them of the second project, I tried to merge the second project with the first one. I did this by "Merge projects" from the file menu. After merging, the new (now third) project appears to not exactly merge the tracks. There should only be two tracks total (animal1 and animal2). Now, there appears to be four different tracks in the whole project. I even gave the tracks the same name in the predictions project. However, I have two "animal1"s and two "animal2"s that are color coded by four different colors. How may I fix this issue so that merging will merge the edits while ensuring that the two tracks are edited rather than making different and additional tracks?

**Comment by roomrys** [comment, marked as answer]
Hi @djg9324,

This issue of handling tracks while merging is being tracked [here](https://github.com/talmolab/sleap/issues/1083). There is currently no way to handle this in the GUI, so it would need to be done in the API. 

I've created an [unofficial gist](https://gist.github.com/roomrys/3d03a68471ce687f5431e7886f55c2ea) to handle this programmatically. You would just need to make edits where there are `TODO`s. The code should be run in your `sleap` environment 
>To activate your environment
>```
>conda active sleap
>```

by calling the file
```bash
python interactive_track_replacement.py
```
and interacts with the user through the terminal. I've minimally tested this to smooth it out.

Let me know if you decide to try it out and/or run into problems. Always save to a new slp!

Thanks,
Liezl

---

An example interaction:
```bash
(sleap) λ python interactive_track_replacement.py

INFO:__main__:All tracks:
        0: Track(spawned_on=0, name='female')
        1: Track(spawned_on=0, name='male')
        2: Track(spawned_on=0, name='pup shaved')
        3: Track(spawned_on=0, name='pup unshaved')

Which tracks do you want to replace? Enter the 'current index':'replacement index' separated by commas. E.g. to replace track 0 with track 2 and track 1 with track 3 we would enter 0:2,1:3. Press enter to continue or q to exit.
0:1,2:3

INFO:__main__:Replacing track 0 with track 1
INFO:__main__:Replacing track 2 with track 3
INFO:__main__:Saved labels to replaced_tracks.slp
INFO:__main__:Opening labels in GUI...
```
